api/app/applicationModel/api.py

```diff
@@ -118,29 +118,31 @@ def post(self):
         print('SECTION ARGS: ', section_args)
 
         for s in section_args:
+            print("Check s_args:", type(section_args))
             section = Section(
-                app_form.event_id,
+                app_form.id,
```

Comment:
@avishkar58 
I have been struggling with this from the beginning, and thought I solved it. I'm quite certain I am missing a step, or not understanding something. 

Using `event_id` is the only way I've been able to avoid this error:
`IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: section.application_form_id
[SQL: INSERT INTO section (application_form_id, name, description, "order", depends_on_question_id, "key") VALUES (?, ?, ?, ?, ?, ?)]
[parameters: (None, u'Section 1', u'Description of the section', 1, None, None)]
(Background on this error at: http://sqlalche.me/e/13/gkpj)`

Reply:
Ah I see - I think what you're missing is a `db.session.commit()` after you do the `db.session.add(app_form)`. This is because "add" essentially adds to a transaction and only when that transaction is committed do the IDs get assigned. Try that and let me know if it fixes the issue? If it doesn't we can debug it together.

Reply:
That solved the issue! Thank you :D